src/scancode/resource.py

```diff
@@ -1497,26 +1497,75 @@ def get_resource_basic_attributes(_resource_data):
             _is_file = _file_type == 'file'
             return _name, _path, _is_file
 
-        # Create root resource
-        root_name, root_path, root_is_file = get_resource_basic_attributes(root_data)
+        def create_empty_resource_data():
+            # Get fields from the base Resource class and the ScannedResource class
+            base_fields = attr.fields(Resource)
+            resource_fields = attr.fields(self.resource_class)
+            # Create dict of {field: field_default_value} for the dynamically created fields
+            resource_data = {}
+            for field in resource_fields:
+                if field in base_fields:
+                    # We only want the fields that are not part of the base set of fields
+                    continue
+                value = field.default
+                if isinstance(value, attr.Factory):
+                    # For fields that have Factories as values, we set their values to be an
+                    # instance of whatever type the factory makes
+                    value = value.factory()
+                resource_data[field.name] = value
+            return resource_data
+
+        # Create root resource without setting root data just yet. If we run into the root data
+        # while we iterate through `resources_data`, we fill in the data then.
+        sample_resource_path = sample_resource_data.get('path')
+        root_path = sample_resource_path.split('/')[0]
+        root_name = root_path
+        root_is_file = False
+        root_data = create_empty_resource_data()
         root_resource = self._create_root_resource(root_name, root_path, root_is_file, root_data)
 
         # To help recreate the resource tree we keep a mapping by path of any
         # parent resource
         parent_by_path = {root_path: root_resource}
 
-        # resources data MUST be in top-down order
+        def get_or_create_parent(path, parent_by_path):
+            """
+            Return a parent resource for a given `path` from `parent_by_path`.
+
+            If a parent resource for a `path` does not exist in `parent_by_path`, it is created recursively.
+
+            Note: the root path and root Resource must already be in `parent_by_path` or else this
+            function does not work.
+            """
+            parent_path = parent_directory(path).rstrip('/')
+            if parent_path in parent_by_path:
```

Comment:
Since this a test that will be done often, it might be best and faster to do instead this where only one dict lookup occurs:
```
existing_parent = parent_by_path.get(parent_path)
if existing_parent:
    return existing_parent
```